### PR TITLE
Lifecycle Events v1

### LINKED ISSUE #22: Server player events
Player leave, player join, player send message, etc. Funni ban players from mod go brrr

### PR DESCRIPTION
**Pull request by valoeghese**
Closes #22, and finalises API 1.1.0

### CODE REVIEW

File: legacy-lifecycle-events-v1/src/test/java/io/github/minecraftcursedlegacy/test/LifecycleEventsTestServer.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import io.github.minecraftcursedlegacy.api.event.lifecycle.ServerLifecycleEvents;
+import net.fabricmc.api.DedicatedServerModInitializer;
+
+public class LifecycleEventsTestServer implements DedicatedServerModInitializer {
+	@Override
+	public void onInitializeServer() {
+		ServerLifecycleEvents.PLAYER_LOGIN.register((player, packetHandler) -> {
+			System.out.println("Player has joined: " + player.name);
+			//packetHandler.drop("U R Bad"); Uncomment this line to test login packet dropping
```
Comment (CoolMineman) [inline legacy-lifecycle-events-v1/src/test/java/io/github/minecraftcursedlegacy/test/LifecycleEventsTestServer.java:34]: bruh

**Review by CoolMineman — APPROVED** [review]

**Review by williambl — APPROVED** [review]

**Review by williambl — APPROVED** [review]
putting unrelated changes in pr irrit

File: legacy-lifecycle-events-v1/src/test/java/io/github/minecraftcursedlegacy/test/LifecycleEventsTestServer.java (hunk 1 above)
Comment (valoeghese) [reply legacy-lifecycle-events-v1/src/test/java/io/github/minecraftcursedlegacy/test/LifecycleEventsTestServer.java:34]: yes

**Comment by valoeghese** [comment]
time to fix up texture api compileTestJava
it's not actually a problem in game of course but I'd rather gh actions doesn't sh1t itself

**Pull request merged**